sigal/plugins/compress_assets.py

```diff
@@ -0,0 +1,153 @@
+        try:
+            file_stats = os.stat(filename)
+            compressed_stats = os.stat(compressed_filename)
+        except OSError:  # FileNotFoundError is for Python3 only
+            pass
+
+        if file_stats and compressed_stats:
+            return compressed_filename if file_stats.st_mtime > compressed_stats.st_mtime else False
+        else:
+            return compressed_filename
+
+
+class GZipCompressor(BaseCompressor):
+    suffix = 'gz'
+
+    def do_compress(self, filename, compressed_filename):
+        with open(filename, 'rb') as f_in, gzip.open(compressed_filename, 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out)
+
+
+class ZopfliCompressor(BaseCompressor):
+    suffix = 'gz'
+
+    def do_compress(self, filename, compressed_filename):
+        import zopfli.gzip
+        with open(filename, 'rb') as f_in, open(compressed_filename, 'wb') as f_out:
+            f_out.write(zopfli.gzip.compress(f_in.read()))
+
+
+class BrotliCompressor(BaseCompressor):
+    suffix = 'br'
+
+    def do_compress(self, filename, compressed_filename):
+        import brotli
+        with open(filename, 'rb') as f_in, open(compressed_filename, 'wb') as f_out:
+            f_out.write(brotli.compress(f_in.read(), mode=brotli.MODE_TEXT))
+
+
+def get_compressor(settings):
+    name = settings.get('method', DEFAULT_SETTINGS['method'])
+    if name == 'gzip':
+        return GZipCompressor(settings)
+    elif name == 'zopfli':
+        try:
+            import zopfli.gzip
+            return ZopfliCompressor(settings)
+        except ImportError:
+            logging.warning('Zopfli not found, using standard gzip')
+            return GZipCompressor(settings)
+
+    elif name == 'brotli':
+        try:
+            import brotli
+            return BrotliCompressor(settings)
+        except ImportError:
+            logger.error('Unable to import brotli module')
+
+    else:
+        logger.error('No such compressor {}'.format(name))
+
+
+def compress_assets(assets_directory, compressor):
+    assets = []
+    for current_directory, _, filenames in os.walk(assets_directory):
+        for filename in filenames:
+            assets.append(os.path.join(current_directory, filename))
+
+    with progressbar(assets, label='Compressing theme assets') as progress_compress:
+        for filename in progress_compress:
+            compressor.compress(filename)
+
+
+def compress_gallery(gallery):
+    logging.info('Compressing assets for %s', gallery.title)
+    compress_settings = gallery.settings.get('compress_assets_options', DEFAULT_SETTINGS)
+    compressor = get_compressor(compress_settings)
+
+    if compressor is None:
+        return
+
+    with progressbar(gallery.albums.values(), label='Compressing albums static files') as progress_compress:
+        for album in progress_compress:
+            compressor.compress(os.path.join(album.dst_path, album.output_file))
```

Comment:
I think you could combine the two progress bars, one is enough for me. Thoughts ?

Reply:
Yes, this is a good idea. I'll have to watch how to do it efficiently. But yes, this is clearly a better UI.

Reply:
If you merge `compress_assets` in `compress_gallery`, then:
```
with progressbar(length=len(gallery.albums) + len(assets)) as bar:
    for album in gallery.albums.values():
        ...
        bar.update()
    for filename in progress_compress:
        ...
        bar.update()
```